observer/observer_test.go

```diff
@@ -56,3 +54,98 @@ func TestObserver_IsConcurrentJobsLimitReached(t *testing.T) {
 	isLimitReached = o.IsConcurrentJobsLimitReached(v1alpha1.ArchiveType, 0)
 	assert.False(t, isLimitReached)
 }
+
+func TestObserver_AreCallbacksInvoked(t *testing.T) {
+	tests := []struct {
+		name                string
+		eventType           EventType
+		registeredCallbacks int
+		expectedInvocations int
+	}{
+		{
+			name:                "succeeded status",
+			eventType:           Suceeded,
+			registeredCallbacks: 1,
+			expectedInvocations: 1,
+		},
```

Comment:
I propose to use the *Given_When_Then* syntax. It's easier to read in the Test spec what is expected or not than the comments in the test logic that are the same for every case.
and make the test cases using a map[string]struct instead of a struct slice, it looks a little bit nicer IMO. See https://github.com/vshn/k8up/blob/v1.0.0-rc1/handler/schedule_test.go#L163 for inspiration

Reply:
Thank you for this input.

I've reworked the tests and split them into two separate tests, because the original code was testing two different things anyway: Are Callbacks invoked per Status, and are all registered callbacks invoked.

These are now two tests and I believe it's now simpler to understand what they do.